### ISSUE TITLE
[Bug] sharing photos from the gallery not working on ios

### ISSUE THREAD

**Issue by dm82m** [schlagmichdoch/PairDrop#336]
**Describe the bug**
It is currently not possible to send files from the photos gallery. If you pick them, the browser reloads on the iphone and nothing is transferred. Transfer is working if you capture a photo in the dialog or send a file attachment.

**To Reproduce**
1. Click on the device on your iOS
2. Choose "Photogallery"
3. Select an image and try to send it
4. "Preparation running..."
5. "This Website was reloaded due to a problem."

**Expected behavior**
Sending of photos from the gallery of the iPhone is working.

**Screenshots**
No real screenshots available.

**Desktop (please complete the following information):**
 - OS: MacOS 15
 - Browser chrome and safari -> both browsers same result!
 - Version up2date on both

**Smartphone (please complete the following information):**
 - Device: iPhone15
 - OS: iOS 18
 - Browser stock browser, safari
 - Version up2date

**Bug occurs on official PairDrop instance https://pairdrop.net/**
Yes
Version: v1.10.10

**Comment by schlagmichdoch** [comment]
Thanks for your report! As I cannot reproduce this on an iOS 17 iPhone I need some more info:
- Has this only happened since you updated your iPhone to iOS 18?
- Is it possible to send images with this shortcut: https://routinehub.co/shortcut/13990/
- How big is the photo you are trying to send?
- Have you tried different types of photos? Screenshots vs. camera shots vs. videos etc.?
- If this is only present in some photos, would you be able to send me a copy of it via mail or drive link? (sonnig-02.hieven@icloud.com)

**Comment by dm82m** [comment]
> * Has this only happened since you updated your iPhone to iOS 18?
I am new to PairDrop, haven't tested it before iOS 18 and I have no iOS < 18 device at hand

> * Is it possible to send images with this shortcut: https://routinehub.co/shortcut/13990/
yes its possible, but it is converting heic to jpg; so no option for me as I want to transfer original files

> * How big is the photo you are trying to send?
1.4MB

> * Have you tried different types of photos? Screenshots vs. camera shots vs. videos etc.?
Problem occurs with .heic files, .jpg are working without an issue

> * If this is only present in some photos, would you be able to send me a copy of it via mail or drive link? ([sonnig-02.hieven@icloud.com](mailto:sonnig-02.hieven@icloud.com))
I sent you one of the images where the transfer is not working

fun fact: using snapdrop.net and its working; but honestly I would prefer to work with pairdrop.net, that's why I created the issue here... ;-)

**Comment by schlagmichdoch** [comment]
Thanks for the infos and quick response! I can reproduce this now, when I share heic images as original instead of „most compatible“.

Apparently, the library used to create thumbnails of heic images is broken and the error is not handled properly. The library seems to crash with some photos only but I’m not sure yet what’s the issue. Here’s a demo page for testing: https://alexcorvi.github.io/heic2any/#try.

I’m short of time atm but I’ll take a look into it and will fix this soon together with #332.

**Comment by schlagmichdoch** [comment]
@dm82m I've just tried again and somehow the problem is not reproducible anymore for me. Is this still an issue for you?

**Comment by dm82m** [comment]
yes it still occurs. I tested it with a new image created today, just 2mb in size and it worked. but as soon as I take heic images that are ~4mb in size, the issue occurs. also with the image I have sent you by mail.

**Comment by dm82m** [comment]
Any progress here?

**Comment by schlagmichdoch** [comment]
You can confirm that this is indeed solved in the underlying library on this demo page:
https://strukturag.github.io/libheif/

**Comment by dm82m** [comment]
> You can confirm that this is indeed solved in the underlying library on this demo page: https://strukturag.github.io/libheif/

yes, it works there. so just waiting for a fix in pairdrop, but its not super urgent for me.

**Comment by schlagmichdoch** [comment]
Thanks for your patience!

I have since implemented this into a test instance. Please verify that everything works for you on this instance: https://pairdrop-next.onrender.com/ before I deploy this as part of the new version!

**Comment by dm82m** [comment]
yes it works, tested the current version -> not working, tested this version here -> working! please let me know when it is deployed to public! and thanks for your work!

**Comment by dm82m** [comment]
@schlagmichdoch when will it be deployed to production?

**Comment by schlagmichdoch** [comment]
I'm really sorry for the delay, I've been really short on time lately.
Sadly, the current development version has still issues. Mainly, the image conversion is running in the main thread which is blocked for the time it takes to convert the file. This leads to unwanted consequences so I need to rewrite it to run in a service worker. This is more complicated than I thought as some more elements should be handled this way which I will include in the restructuring.

To prevent the page from crashing until the PR is finished, I just deployed a hotfix that simply deactivates the thumbnail creation for HEIC/HEIF files altogether, which is what I should have done as soon as this emerged. This way PairDrop should work for you again, albeit without thumbnail creation on iOS.
